enclave_wrangler/objects_api.py

```diff
@@ -351,60 +357,44 @@ def fetch_cset_and_member_objects(
-        #       processed: timestamp, initially null # used by code that updates the derived tables
-
         version: int = cset['properties']['codesetId']
-        # Hierarchical
-        # TODO: exc handling: Will need to handle differently or do more: https://github.com/jhu-bids/TermHub/issues/451
         try:
-            cset['expression_items']: List[Dict] = get_concept_set_version_expression_items(version, return_detail='full')
+            cset['expression_items']: List[Dict] = \
+                get_concept_set_version_expression_items(version, return_detail='full')
         except EnclavePaginationLimitErr as err:
             cset['expression_items']: List[Dict] = err.args[1]['results_prior_to_error']
-            # TODO: flag in some table that we need to do datasets api download
-            # cset['termhub-notes']['expression_items_failed_after'] = len(cset['expression_items'])
-            print(pagination_err.format(version), file=sys.stderr)
-
+            fetch_statuses.append({'table': 'code_sets', 'primary_key': version, 'status_initially':
+                'fail-excessive-items', 'comment': f"Failed after {len(cset['expression_items'])} expression_items."})
+        except Exception as err:
+            insert_fetch_status(fetch_statuses)
+            raise err
         try:
             cset['member_items']: List[Dict] = get_concept_set_version_members(version, return_detail='full')
-            auditMsg = f"found {len(cset['member_items'])} members"
         except EnclavePaginationLimitErr as err:
             cset['member_items']: List[Dict] = err.args[1]['results_prior_to_error']
-            print(pagination_err.format(version), file=sys.stderr)
-            auditMsg = f"failed after {len(cset['member_items'])} members"
-
-        if not cset['member_items']:
-            auditMsg = f"got no members as of {str(datetime.now())}, will continue trying until..."
-            pass
-            # TODO: do something so we keep trying for a couple hours. it's possible it really doesn't have members, but
-            #   probably the members just aren't available yet
-
-
+            fetch_statuses.append({'table': 'code_sets', 'primary_key': version, 'status_initially':
+                'fail-excessive-members', 'comment': f"Failed after {len(cset['member_items'])} members."})
+        except Exception as err:
+            insert_fetch_status(fetch_statuses)
+            raise err
+        # todo: consider adding `and cset['expression_items']:` to further avoid false positives, or is the isDraft
+        #  check sufficient? I think it's correct that items are available immediately upon creation, but members are
+        #  not? Can a draft be finalized and have 0 expression items? Another thing to check would be createdAt >1+hours
+        #  as proxy; e.g. 'createdAt': '2023-02-14T15:10:13.921Z',
+        if not cset['member_items'] and not cset['properties']['isDraft']:
```

Comment:
@Sigfried Regarding the issue where a cset is new and thus has 0 members when we fetch it:

I found an edge cases where there are 0 members, but it's OK and we don't need to flag:
1.  The cset has no members, but is a draft

Also, there are 2 other non-problematic cases I thought of that I could also check for in this if statement:
2. The cset has no members, but also has no expression items (_@Sigfried I think this can't happen, so I didn't add logic to check for it, but I will add if it can. Can it?)_
3. The cset has no members, but was created 'x' hours ago (_@Sigfried this is another way we can check things, but I thought it's better not to add this, as '1' and '2' should be sufficient to catch cases._)